static/app/views/dashboardsV2/utils.tsx

```diff
@@ -409,19 +409,27 @@ export function hasUnsavedFilterChanges(
   newDashboardFilters: DashboardFilters
 ) {
   const savedFilters = {
-    projects: initialDashboard.projects,
-    environment: initialDashboard.environment,
+    projects: new Set(initialDashboard.projects),
```

Comment:
I feel like manually identifying which values are arrays isn't the best experience for extending page filters in the future.

I was thinking of using lodash's `isEqualWith` utility to define a custom checker, but I felt like this was the simpler and clearer solution at the time. If anyone feels strongly about doing that work then I can take a look but since we have a nested object (`filters` with `release`), we might need something recursive or just define a separate function for comparing the `filters` object